pkg/output/kafka/kafka_test.go

```diff
@@ -0,0 +1,120 @@
+// Licensed to Elasticsearch B.V. under one or more agreements.
+// Elasticsearch B.V. licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+package kafka
+
+import (
+	"log"
+	"net"
+	"os"
+	"os/signal"
+	"strings"
+	"testing"
+
+	"github.com/Shopify/sarama"
+	"github.com/ory/dockertest/v3"
+	"github.com/ory/dockertest/v3/docker"
+	"github.com/stretchr/testify/require"
+	"gotest.tools/assert"
+
+	"github.com/elastic/stream/pkg/output"
+)
+
+const (
+	emulatorHost = "127.0.0.1"
+	emulatorPort = "9092"
+	topic        = "testTopic"
+)
+
+var (
+	outputevent         string
+	emulatorHostAndPort = net.JoinHostPort(emulatorHost, emulatorPort)
+)
+
+func TestMain(m *testing.M) {
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		log.Fatalf("Could not connect to docker: %s", err)
+	}
+
+	resource, err := pool.RunWithOptions(&dockertest.RunOptions{
+		Repository: "bashj79/kafka-kraft",
+		Tag:        "latest",
+		PortBindings: map[docker.Port][]docker.PortBinding{
+			emulatorPort: {{HostIP: emulatorHost, HostPort: emulatorPort}},
```

Comment:
We ended up not implementing this after some discussion, since the broker announces the port in use, it has to match the local port for this to easily work.